blockade/api/rest.py

```diff
@@ -229,9 +230,84 @@ def destroy(name):
     if not BlockadeManager.blockade_exists(name):
         abort(404)
 
+    if _chaos.exists(name):
+        try:
+            _chaos.delete(name)
+        except errors.BlockadeUsageError as bue:
+            app.logger.error(bue)
+
     b = BlockadeManager.get_blockade(name)
     b.destroy()
 
     BlockadeManager.delete_config(name)
 
     return '', 204
+
+
+_chaos = chaos.Chaos()
+
+
+def _validate_chaos_input(option):
+    valid_inputs = [
+        "min_start_delay", "max_start_delay",
+        "min_run_time", "max_run_time",
+        "min_containers_at_once", "max_containers_at_once",
+        "min_events_at_once", "max_events_at_once",
+        "event_set"
+    ]
+    for o in option:
+        if o not in valid_inputs:
+            raise errors.BlockadeHttpError(400, "%s is not a valid input")
+
+
+@app.route("/blockade/<name>/chaos", methods=['POST'])
+def chaos_new(name):
+    if not BlockadeManager.blockade_exists(name):
+        abort(404, "The blockade %s does not exist" % name)
+    if not request.headers['Content-Type'] == 'application/json':
+        abort(415, "The body is not in JSON format")
+    options = request.get_json()
+    _validate_chaos_input(options)
+    try:
+        _chaos.new_chaos(name, **options)
+        return "Successfully started chaos on %s" % name, 201
+    except errors.BlockadeUsageError as bue:
+        return bue.http_msg, bue.http_code
```

Comment:
You log the exception on destroy and status, but looks like maybe missed here and on updating the options.